### PR TITLE
LTD-3290 Timeout on TAU product assessment

### PR DESCRIPTION
**Pull request by Tllew**
removed current_object and changed the goodonapplication query to point at good.id, the frontend changes reflect this also

[LTD-3290](https://uktrade.atlassian.net/jira/software/c/projects/LTD/boards/204?modal=detail&selectedIssue=LTD-3290)

[LTD-3290]: https://uktrade.atlassian.net/browse/LTD-3290?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/goods/views.py [hunk 1]
```diff
@@ -227,7 +231,6 @@ def post(self, request, case_pk):
             else:
                 good.flags.remove(SystemFlags.WASSENAAR)
 
-        apply_good_flagging_rules_for_case(case)
```
Comment (currycoder) [inline api/goods/views.py:230]: IMO if we can switch to a bulk endpoint here for a workable speed gain, we shouldn't remove this at the same time.  I'm still uncomfortable removing this without a stack of extra testing/thinking.

**Review by depsiatwal — APPROVED** [review]
lgtm

File: api/goods/tests/test_control_codes.py [hunk 2]
```diff
@@ -505,7 +540,7 @@ def test_report_summary_updates_same_product_added_twice(self):
         self.assertEqual(self.product_on_application1.report_summary, "Sniper rifles (10)")
         self.assertEqual(self.product_on_application2.report_summary, "Rifles (5)")
         audit_qs = Audit.objects.filter(verb=AuditType.PRODUCT_REVIEWED)
-        self.assertEqual(audit_qs.count(), 3)
+        self.assertEqual(audit_qs.count(), 1)
         # because we added the same product twice check if reviewing the second has not modified
         # first product report summary value
```
Comment (kevincarrogan) [inline api/goods/tests/test_control_codes.py:545]: Is this comment relevant anymore? Are there tests later that might need to change as well?

Comment (Tllew) [reply api/goods/tests/test_control_codes.py:545]: removed the comment,

I've also added some changes to tests to cover the good.good.id cases

**Pull request merged**